File: 1436-destination-city/1436-destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
        
        Map<String, String> cityMap = new HashMap<>();

        for (List<String> path : paths) {
            cityMap.put(path.get(0), path.get(1));
        }

        
        String destination = paths.get(0).get(1); 
        while (cityMap.get(destination) != null) {
            destination = cityMap.get(destination);
        }
        
        return destination;
        
    }
}

File: 1436-destination-city/1436-destination-city.kt
class Solution {
fun destCity(paths: List<List<String>>): String {
    val map = paths.map { it[0] to it[1] }.toMap()
    var key = paths[0][0]
    while (map.containsKey(key)) {
        key = map[key]!!
    }
    return key
}
}